src/Webfactory/Slimdump/SlimdumpCommand.php

```diff
@@ -56,10 +58,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
     private function connect($dsn)
     {
         try {
-            return \Doctrine\DBAL\DriverManager::getConnection(
-                array('url' => $dsn, 'charset' => 'utf8', 'driverClass' => 'Doctrine\DBAL\Driver\PDOMySql\Driver')
+            /**
+             * a dsn string containing mysqli will result in the DriverManager creating a MysqliConnection but we need PDO
+             * @see DriverManager::parseDatabaseUrlScheme()
+             */
+            $mysqliIndependentDsn = str_replace('mysqli://', 'mysql://', $dsn);
```

Comment:
Would it be more cautious to use a `preg_replace('_^mysqli://_', ...)`?

Reply:
Ah yes I like that beginning of string check!